src/main/java/de/eldoria/pickmeup/listener/CarryListener.java

```diff
@@ -92,57 +92,113 @@ public void onEntityInteract(PlayerInteractAtEntityEvent event) {
     @EventHandler
     public void onSneak(PlayerToggleSneakEvent event) {
         Player player = event.getPlayer();
-        if (!mountStates.containsKey(player.getUniqueId())) {
+        UUID playerUUID = player.getUniqueId();
+        MountState currentState = mountStates.get(playerUUID);
+
+        if (currentState == null) {
             unmountAll(player);
             return;
         }
 
-        MountState mountState = mountStates.get(player.getUniqueId());
+        if (!event.isSneaking()) {
+            handleSneakRelease(player, playerUUID, currentState);
+        } else {
+            handleSneakPress(player, playerUUID, currentState);
+        }
```

Comment:
Better code readability. If you do if else avoid negations, since those are easily overlooked

```suggestion
        if (event.isSneaking()) {
            handleSneakPress(player, playerUUID, currentState);
        } else {
            handleSneakRelease(player, playerUUID, currentState);
        }
```